bluesky_browser/frameworks/qt/search.py

```diff
@@ -163,9 +164,29 @@ def __del__(self):
     def list_subcatalogs(self):
         self._subcatalogs.clear()
         self.catalog_selection_model.clear()
+        event = threading.Event()
+
+        def test_and_add(name, event):
+            # Check that the catalog can be opened.
+            try:
+                self.catalog[name]()
+            except Exception:
+                log.exception(f"Error opening subcatalog {name}")
+            else:
+                self._subcatalogs.append(name)
+                self.catalog_selection_model.appendRow(QStandardItem(str(name)))
+                event.set()
+
         for name in self.catalog:
-            self._subcatalogs.append(name)
-            self.catalog_selection_model.appendRow(QStandardItem(str(name)))
+            self.executor.submit(test_and_add, name, event)
+
+        # When at least one catalog has been confirmed to be open-able,
+        # `event` will be set and we will set that as the default catalog.
+        while True:
+            if event.wait(timeout=5):
```

Comment:
should we return as soon as we find _any_ catalog?  That is enough to populate the UI.

Reply:
That's the intent. The first Catalog we find sets the Event. Have I implemented something different here?

Reply:
Why do we have a `while` loop then?

Reply:
We wait `timeout` seconds and then log to inform the user where the holdup is occurring before going back to waiting. We keep doing this every `timeout` seconds.

This will all be moot when we fix this properly---see my response to @dylanmcreynolds below.